### ISSUE TITLE
Spec 0014: Unified Skin Diagnostics Core, Clinical Dossier Engine, and Mobile Offline Store

### ISSUE THREAD

**Issue by dinh144** [dinh144/skinsense#36]
# Spec 0014: Unified Skin Diagnostics Core, Clinical Dossier Engine, and Mobile Offline Store

## Problem Statement

SkinSense has transitioned into a 100% Autonomous AI Dermo-Pharmacy and Skincare Routine Platform (ADR 0036). Following the successful purge of legacy telemedicine routes and the deepening of `DermoPharmacistEngine` (Spec 0013), two remaining architectural friction points exist across the Web and Mobile codebases:

1. **Fragmented Computer Vision & Triage Pipeline (Web)**: Skin lesion screening is fragmented across 5 shallow helper files (`analysis.ts`, `preflight.ts`, `triage.ts`, `real-vision-pipeline.ts`, `shared.ts`). Consumers like `/api/skin/analyze` and the `/analyze` UI page must manually coordinate preflight blur/lighting checks, model tensor loading, 7-class DermaMNIST softmax calculations, and ABCDE triage classification.
2. **Coupled Dossier PDF Generation (Web)**: Clinical referral PDF generation is tightly coupled to React DOM modal components (`ClinicalDossierModal.tsx`), preventing clean server-side, background, or headless generation of hospital export dossiers.
3. **Leaky Mobile Offline SQLite Seam (Mobile)**: Mobile UI screens (Routine Screen, Catalog Screen) directly interact with SQLite repositories and manually trigger `SyncCoordinator` queues, creating state leakages and synchronization race conditions.

## Solution

1. **Deepen `SkinDiagnosticsCore`**: Consolidate canvas preflight verification, DermaMNIST 7-class CNN inference, ABCDE risk triage, and hospital referral gates into a single deep module with one public entry point: `SkinDiagnosticsCore.evaluateSkinLesion(input, profile): Promise<DiagnosticDossier>`.
2. **Deepen `ClinicalDossierEngine`**: Decouple hospital-grade PDF export generation from React DOM, exposing a pure, headless PDF generation engine: `ClinicalDossierEngine.generateDossierPDF(data)`.
3. **Deepen `OfflinePharmacyStore`**: Consolidate mobile SQLite operations and Cloud Sync queues into a clean reactive hooks adapter: `OfflinePharmacyStore` (`useLiveRoutine`, `useLiveCatalog`, `useSyncStatus`).

---

## User Stories

1. As a patient, I want to upload a lesion photo and receive an immediate end-to-end evaluation (image quality check, DermaMNIST 7-class prediction, risk triage level, and matching INCI actives) without server errors or UI lag.
2. As a patient with a suspicious lesion, I want to export a 1-click hospital-grade clinical dossier PDF containing ABCDE criteria, lesion evolution timeline, and high-risk referral warnings to bring to a dermatologist.
3. As a mobile app user, I want my skincare routine steps and catalog browsing to work 100% offline with zero lag, automatically syncing in the background when connectivity resumes.
4. As a web API developer, I want `/api/skin/analyze` to call a single high-leverage method `SkinDiagnosticsCore.evaluateSkinLesion()` rather than juggling 5 shallow helpers.
5. As a mobile engineer, I want clean reactive hooks (`useLiveRoutine`, `useLiveCatalog`) that encapsulate all SQLite transactions, conflict resolution, and dirty state tracking behind a single seam.
6. As a clinical auditor, I want high-risk melanoma (`mel`), basal cell carcinoma (`bcc`), and actinic keratosis (`akiec`) predictions to automatically trigger the `HIGH_RISK_URGENT` triage gate with hospital referral notes.
7. As a dark skin phototype patient (Fitzpatrick IV-VI), I want post-inflammatory hyperpigmentation (PIH) warnings to be automatically injected into the diagnostic dossier.
8. As an operations lead, I want 100% unit test coverage over the unified diagnostics, dossier, and mobile offline store modules to ensure zero regression across releases.

---

### 1. Seam Architecture & Deep Modules

- **`SkinDiagnosticsCore` (`apps/web/lib/skin/skin-diagnostics-core.ts`)**:
  - Encapsulates:
    - Preflight sharpness (Laplacian variance) and lighting exposure validation.
    - 7-class DermaMNIST CNN tensor normalization and softmax inference (`akiec`, `bcc`, `bkl`, `df`, `mel`, `nv`, `vasc`).
    - ABCDE clinical criteria and 3-tier triage level computation (`LOW`, `MODERATE`, `HIGH`).
    - INCI active ingredient recommendations and hospital referral gate.
  - Public Interface:
    ```ts
    export interface LesionInput {
      imageBytes?: Uint8Array;
      imageData?: ImageData | { data: Uint8ClampedArray; width: number; height: number };
      mimeType?: string;
    }

    export interface PatientProfile {
      fitzpatrickScale?: number;
      bodyLocation?: string;
      isEvolving?: boolean;
      isBleeding?: boolean;
    }

    export class SkinDiagnosticsCore {
      static async evaluateSkinLesion(input: LesionInput, profile?: PatientProfile): Promise<DiagnosticDossier>;
      static validateImageQuality(imageData: { data: Uint8ClampedArray; width: number; height: number }): PreflightResult;
    }
    ```

- **`ClinicalDossierEngine` (`apps/web/lib/skin/clinical-dossier-engine.ts`)**:
  - Headless clinical dossier document synthesizer generating formatted HTML/PDF payloads for hospital handoff.
  - Public Interface:
    ```ts
    export class ClinicalDossierEngine {
      static generateHospitalDossierData(dossier: DiagnosticDossier, patientInfo: { name: string; age?: number }): HospitalDossierReport;
    }
    ```

- **`OfflinePharmacyStore` (`apps/mobile/lib/store/offline-pharmacy-store.ts`)**:
  - Reactive adapter wrapping SQLite transactions and `SyncCoordinator` queues for Expo / React Native.
  - Public Interface:
    ```ts
    export class OfflinePharmacyStore {
      static useLiveRoutine(userId: string): { routine: RoutineItem[]; markStepDone: (id: string) => Promise<void> };
      static useLiveCatalog(): { catalog: Product[]; searchINCI: (q: string) => Product[] };
      static useSyncStatus(): { isSyncing: boolean; pendingChangesCount: number; triggerManualSync: () => Promise<void> };
    }
    ```

---

## Testing Decisions

- **Black-Box Testing on Highest Seams**:
  - Test `SkinDiagnosticsCore.evaluateSkinLesion` for preflight failure on blank/blurry images, melanoma triage escalation on high logits, and benign nevus classification on typical moles.
  - Test `ClinicalDossierEngine.generateHospitalDossierData` for inclusion of ABCDE criteria, medical disclaimer, and triage tier badge.
  - Test `OfflinePharmacyStore` for optimistic routine step completion, offline queue ingestion, and sync state transitions.
- **Monorepo Build Verification**: Ensure all 197+ unit tests pass, typecheck produces 0 errors, and Next.js static prerender generates 25 clean production routes.

---

## Out of Scope

- Native iOS/Android biometric face scanning (FaceID / TouchID).
- Real-time video stream lesion segmentation.
- External hospital EHR HL7/FHIR server integration.

---

## Further Notes

- Fully adheres to **ADR 0036** and **ADR 0045**.

**Comment by dinh144** [comment]
Completed: Spec 0014 fully implemented, verified, and certified across monorepo.
